lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -144,20 +144,24 @@ function isObjectLocked(bucket, objectMD, headers) {
     return false;
 }
 
-function validateObjectLockUpdate(objectMD, retentionInfo) {
+function validateObjectLockUpdate(objectMD, retentionInfo, bypassGovernance) {
     const { retentionMode: existingMode, retentionDate: existingDateISO } = objectMD;
-    if (!existingMode || existingMode === 'GOVERNANCE') {
+    if (!existingMode) {
         return null;
     }
 
     const existingDate = new Date(existingDateISO);
     const isExpired = existingDate < Date.now();
 
-    if (retentionInfo.mode === 'GOVERNANCE' && !isExpired) {
+    if (existingMode === 'GOVERNANCE' && !isExpired && !bypassGovernance) {
         return errors.AccessDenied;
     }
 
-    if (new Date(retentionInfo.date) < existingDate) {
+    if (existingMode === 'COMPLIANCE' && retentionInfo.mode === 'GOVERNANCE' && !isExpired) {
```

Comment:
Just for clarity, could benefit to group compliance checks in a block, or maybe even also grouping governance checks with a switch-like:

```javascript
if (existingMode === 'GOVERNANCE') {
    // governance checks
} else if (existingMode === 'COMPLIANCE') {
    // compliance checks
}
```